core/src/main/java/hudson/util/ProcessKillingVeto.java

```diff
@@ -47,7 +49,7 @@
  * 
  * @author <a href="mailto:daniel.weber.dev@gmail.com">Daniel Weber</a>
  */
-public abstract class ProcessKillingVeto implements ExtensionPoint {
+public abstract class ProcessKillingVeto implements ExtensionPoint, Serializable {
```

Comment:
Wouldn't that approach require going back to the master for every process? Is that really needed or a good idea?

Well, I think you'd probably only go back for those that have the cookie/env. variable set, so maybe not so bad.